lib/internal/crypto/dsa.js

```diff
@@ -0,0 +1,273 @@
+        hash: { name: hash.name }
+      };
+
+      const publicKey =
+        new CryptoKey(
+          pubKey,
+          algorithm,
+          getUsagesUnion(usageSet, 'verify'),
+          true);
+
+      const privateKey =
+        new CryptoKey(
+          privKey,
+          algorithm,
+          getUsagesUnion(usageSet, 'sign'),
+          extractable);
+
+      resolve({ publicKey, privateKey });
+    });
+  });
+}
+
+function dsaExportKey(key, format) {
+  return jobPromise(new DSAKeyExportJob(
+    kCryptoJobAsync,
+    format,
+    key[kKeyObject][kHandle]));
+}
+
+async function dsaImportKey(
+  format,
+  keyData,
+  algorithm,
+  extractable,
+  keyUsages) {
+  const { hash } = algorithm;
+  if (hash === undefined)
+    throw new ERR_MISSING_OPTION('algorithm.hash');
+
+  const usagesSet = new SafeSet(keyUsages);
+  let keyObject;
+  switch (format) {
+    case 'node.keyObject': {
+      if (!isKeyObject(keyData))
+        throw new ERR_INVALID_ARG_TYPE('keyData', 'KeyObject', keyData);
+      if (keyData.type === 'secret')
+        throw lazyDOMException('Invalid key type', 'InvalidAccessException');
+      verifyAcceptableDsaKeyUse(algorithm.name, keyData.type, usagesSet);
+      keyObject = keyData;
+      break;
+    }
+    case 'spki': {
+      verifyAcceptableDsaKeyUse(algorithm.name, 'public', usagesSet);
+      keyObject = createPublicKey({
+        key: keyData,
+        format: 'der',
+        type: 'spki'
+      });
+      break;
+    }
+    case 'pkcs8': {
+      verifyAcceptableDsaKeyUse(algorithm.name, 'private', usagesSet);
+      keyObject = createPrivateKey({
+        key: keyData,
+        format: 'der',
+        type: 'pkcs8'
+      });
+      break;
+    }
+    case 'jwk': {
+      if (keyData == null || typeof keyData !== 'object')
+        throw lazyDOMException('Invalid JWK keyData', 'DataError');
+
+      verifyAcceptableDsaKeyUse(
+        algorithm.name,
+        keyData.x !== undefined ? 'private' : 'public',
+        usagesSet);
+
+      if (keyData.kty !== 'DSA')
+        throw lazyDOMException('Invalid key type', 'DataError');
+
+      if (usagesSet.size > 0 &&
+          keyData.use !== undefined &&
+          keyData.use !== 'sig') {
+        throw lazyDOMException('Invalid use type', 'DataError');
+      }
+
+      validateKeyOps(keyData.key_ops, usagesSet);
+
+      if (keyData.ext !== undefined &&
+          keyData.ext === false &&
+          extractable === true) {
+        throw lazyDOMException('JWK is not extractable', 'DataError');
+      }
+
+      let hash;
+
+      if (keyData.alg != null) {
+        if (typeof keyData.alg === 'object') {
+          const alg = normalizeAlgorithm(keyData.alg);
+          hash = alg.name;
+        } else {
+          hash = normalizeHashName(
+            keyData.alg,
+            normalizeHashName.kContextWebCrypto);
+        }
+      }
+
+      if (hash !== algorithm.hash.name)
+        throw lazyDOMException('Hash mismatch', 'DataError');
```

Comment:
JWK `alg` member is optional, not required. 

This code makes all imports without `keyObject.alg` fail for RSA JWKs.

Furthermore the code suggests that `keyData.alg` may be an object, which it never is, the `alg` property must either be missing or be a string.